### PR TITLE
Add currentFrame to AudioProcess.process call

### PR DESCRIPTION
**Pull request by orottier**
We now support all three attributes from
https://www.w3.org/TR/webaudio/#AudioWorkletGlobalScope-attributes

### CODE REVIEW

File: src/render/processor.rs [hunk 1]
```diff
@@ -38,7 +38,8 @@ pub trait AudioProcessor: Send {
         inputs: &[AudioRenderQuantum],
         outputs: &mut [AudioRenderQuantum],
         params: AudioParamValues,
-        timestamp: f64,
+        current_frame: u64,
+        current_time: f64,
         sample_rate: SampleRate,
     ) -> bool;
```
Comment (orottier) [inline src/render/processor.rs:44]: My personal limit for number of method arguments is five. We are at six now. I'm wondering if I should combine `current_frame`, `current_time` and `sample_rate` in a `GlobalScope` struct

File: src/render/processor.rs [hunk 2]
```diff
@@ -7,6 +7,18 @@ use crate::SampleRate;
...
 use super::{graph::Node, AudioRenderQuantum, NodeIndex};
 
+#[non_exhaustive] // we may want to add user-provided blobs to this later
+#[derive(Copy, Clone, Debug)]
+/// The execution context of all AudioProcessors in a given AudioContext
+///
+/// This struct currently only contains information about the progress of time. In a future
+/// version, it should be possible to add arbitrary data. For example, multiple processors might
+/// share a buffer defining a wavetable or an impulse response.
+pub struct GlobalScope {
```
Comment (orottier) [inline src/render/processor.rs:17]: I organized the three variables in the GlobalScope now, which is loosely based on the https://www.w3.org/TR/webaudio/#audioworkletglobalscope
These concepts of scopes and worklets don't really translate to rust so this is perhaps the best we can do. With the current setup we leave the future ability open (with no breaking changes) for
> Note: The [AudioWorkletGlobalScope](https://www.w3.org/TR/webaudio/#audioworkletglobalscope) may also contain any other data and code to be shared by these instances. As an example, multiple processors might share an ArrayBuffer defining a wavetable or an impulse response.

**Comment by orottier** [comment]
Ready for review but I want to let this simmer for a few days myself. Not sure where the best balance of spec compliance versus decent Rust paradigms lies

**Comment by b-ma** [comment]
Yup, no real strong opinion on my side. My main point would be to say that:

> With the current setup we leave the future ability open (with no breaking changes) for
> > Note: The [AudioWorkletGlobalScope](https://www.w3.org/TR/webaudio/#audioworkletglobalscope) may also contain any other data and code to be shared by these instances. As an example, multiple processors might share an ArrayBuffer defining a wavetable or an impulse response.

is a very good point!

Just one note: maybe `RenderScope` instead of `Scope` would be a bit more precise

**Comment by orottier** [comment]
> Just one note: maybe `RenderScope` instead of `Scope` would be a bit more precise

Great suggestion, that is a lot more fitting.

I also made a minor but quite important change to hand a reference instead of an owned RenderScope to the processors, see 32213b653fbd0055

**Pull request merged**